packages/bridging/src/providers/near-intents/NearIntentsBridgeProvider.ts

```diff
@@ -0,0 +1,370 @@
+import type { Hex } from 'viem'
+import type {
+  BridgeDeposit,
+  BridgeHook,
+  BridgeProvider,
+  BridgeProviderInfo,
+  BridgeQuoteResult,
+  BridgeStatusResult,
+  BridgingDepositParams,
+  BuyTokensParams,
+  GetProviderBuyTokens,
+  QuoteBridgeRequest,
+} from '../../types'
+
+export interface NearIntentsQuoteResult extends BridgeQuoteResult {
+  depositAddress: Hex
+}
+
+export const NEAR_INTENTS_HOOK_DAPP_ID = `${HOOK_DAPP_BRIDGE_PROVIDER_PREFIX}/near-intents`
+
+export interface NearIntentsBridgeProviderOptions {
+  cowShedOptions?: CowShedSdkOptions
+}
+
+export class NearIntentsBridgeProvider implements BridgeProvider<NearIntentsQuoteResult> {
+  protected api: NearIntentsApi
+  protected cowShedSdk: CowShedSdk
+
+  info: BridgeProviderInfo = {
+    name: 'Near Intents',
+    logoUrl: `${RAW_PROVIDERS_FILES_PATH}/near-intents/near-intents-logo.png`,
+    dappId: NEAR_INTENTS_HOOK_DAPP_ID,
+    website: 'https://www.near.org/intents',
+  }
+
+  constructor(options?: NearIntentsBridgeProviderOptions, _adapter?: AbstractProviderAdapter) {
+    const adapter = _adapter || options?.cowShedOptions?.adapter
+    if (adapter) {
+      setGlobalAdapter(adapter)
+    }
+    this.api = new NearIntentsApi()
+    this.cowShedSdk = new CowShedSdk(adapter, options?.cowShedOptions?.factoryOptions)
+  }
+
+  async getNetworks(): Promise<ChainInfo[]> {
+    return NEAR_INTENTS_SUPPORTED_NETWORKS
+  }
+
+  async getBuyTokens(params: BuyTokensParams): Promise<GetProviderBuyTokens> {
+    const tokens = adaptTokens(await this.api.getTokens())
+    return {
+      tokens: tokens.filter((token) => token.chainId === params.buyChainId),
+      isRouteAvailable: tokens.length > 0,
+    }
+  }
+
+  async getIntermediateTokens(request: QuoteBridgeRequest): Promise<TokenInfo[]> {
+    if (request.kind !== OrderKind.SELL) {
+      throw new BridgeProviderQuoteError(BridgeQuoteErrors.ONLY_SELL_ORDER_SUPPORTED, { kind: request.kind })
+    }
+
+    const { sellTokenChainId, buyTokenChainId, buyTokenAddress } = request
+
+    const tokens = adaptTokens(await this.api.getTokens())
+    const sourceTokens = tokens.filter((token) => token.chainId === sellTokenChainId)
+    const targetTokens = tokens.filter((token) => token.chainId === buyTokenChainId)
+
+    const wrappedTokenAddresses = Object.values(NEAR_INTENTS_BLOCKCHAIN_TO_NATIVE_WRAPPED_TOKEN_ADDRESS).map((a) =>
+      a.toLowerCase(),
+    )
+    const buyTokens = targetTokens.find((token) => {
+      // Supports both the native token and its wrapped version (e.g. POL and WPOL, where POL is represented by ETH_ADDRESS).
+      if (token.address === ETH_ADDRESS || wrappedTokenAddresses.includes(token.address.toLowerCase()))
+        return token.chainId === buyTokenChainId
+      return token.address.toLowerCase() === buyTokenAddress.toLowerCase()
+    })
+    if (!buyTokens) return []
+
+    // If buyToken is supported, all source tokens can be used to buy buyToken
```

Comment:
That's awesome, unfortunately the SDK implementation will only try with the first in the array). So ideally, we have the list sorted by relevance (more liquid ones first).

Future versions might consider more and query the API in parallel to understand which one gives you more tokens.